### ISSUE TITLE
`addGeosFunctions.mjs` should reference docs for correct GEOS version

### ISSUE THREAD

**Issue by chrispahm** [chrispahm/geos-wasm#43]
As @mthh noted, we currently have a bug when running `npm run make`, stating that the function `_GEOSGeom_transformXYZ_r` does not exist. 

The reason this bug appears is that in the `addGeosFunctions.mjs` we use the documentation hosted on https://libgeos.org/doxygen/geos__c_8h.html to build the Emscripten function signatures. The docs are auto-generated from the GEOS `main` branch, so always the latest changes are included even if they are not released yet (docs version is 3.14.0dev, latest Release is 3.13.0). The `_GEOSGeom_transformXYZ_r ` shows up in the docs, but isn't available in 3.13.0, hence the error ☹️  

I couldn't find a version history of the docs on the libgeos.org page, so I think our only option is to include a section in the `Makefile` that generates the doxygen docs. Then we can reference the correct documentation, and the bugs should disappear! 

TLDR: Proposed solution → [add docs to GEOS build steps](https://github.com/libgeos/geos/tree/main?tab=readme-ov-file#using-the-c-interface-1) in our Makefile → reference the generated `geos__c_8h.html` instead of the version online

**Comment by chrispahm** [comment]
Note that https://www.doxygen.nl/index.html is now required as part of the build step
